src/frontend/packages/tdbp/src/pages/Student/index.tsx

```diff
@@ -15,19 +14,25 @@ import { TdbpFiltersProvider } from "../../contexts";
 export default () => {
   const { appData } = useLTIContext();
 
-  if (!appData.course_id) {
+  const tokenPayload = appData.access.split(".");
+  const decodedPayload = JSON.parse(atob(tokenPayload[1]));
```

Comment:
This is what I mentin before, you should decode it once and then make it available everywhere in the application